Foundation/NSFileManager.swift

```diff
@@ -340,17 +340,15 @@ public class NSFileManager : NSObject {
         result[NSFileSystemFileNumber] = NSNumber(unsignedLongLong: UInt64(s.st_ino))
         
         let pwd = getpwuid(s.st_uid)
-        if pwd != nil && pwd.memory.pw_name != nil {
-            if let name = String.fromCString(pwd.memory.pw_name) {
-                result[NSFileOwnerAccountName] = name
-            }
+        if pwd != nil && pwd.pointee.pw_name != nil {
+            let name = String(cString: pwd.pointee.pw_name)
+            result[NSFileOwnerAccountName] = name
```

Comment:
why was this logic changed? couldn't a string from c potentially be corrupted? I have seen that type of failure in C before why can't it happen here? Is this perhaps the wrong API to call if we want a failable C string conversion (without bouncing to CFStringRef/NSString, which can handle a failable conversion)